client_files/ElasticSearch.sh

```diff
@@ -0,0 +1,167 @@
+#
+# This script installs everything required to use elasticsearch in MediaWiki
+# 
+# Dependencies
+# - PHP compiled with cURL
+# - Elasticsearch
+#   - JAVA 7+
+# - Extension:Elastica
+# - Extension:CirrusSearch
+# 
+# Ref:
+# https://www.mediawiki.org/wiki/Extension:CirrusSearch
+# https://en.wikipedia.org/w/api.php?action=cirrus-config-dump&srbackend=CirrusSearch&format=json
+# https://git.wikimedia.org/blob/mediawiki%2Fextensions%2FCirrusSearch.git/REL1_25/README
+# https://www.mediawiki.org/wiki/Extension:CirrusSearch/Tour
+# https://wikitech.wikimedia.org/wiki/Search
+#
+
+#
+# Install JAVA
+# 
+# http://docs.oracle.com/javase/8/docs/technotes/guides/install/linux_jdk.html#BJFJHFDD
+# http://stackoverflow.com/questions/10268583/how-to-automate-download-and-installation-of-java-jdk-on-linux
+#
+echo "******* Downloading and installing JAVA Development Kit *******"
+cd ~/sources/meza1/client_files
+yum -y install java-1.7.0-openjdk
+# Reference this for if we want to try JDK 8: http://tecadmin.net/install-java-8-on-centos-rhel-and-fedora/
+## wget --no-check-certificate --no-cookies --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u45-b14/jdk-8u45-linux-x64.rpm
+## rpm -ivh jdk-8u45-linux-x64.rpm
+
+# Verify JAVA is installed
+java -version
```

Comment:
If java isn't installed what do you do? Should it die? In install.sh a check for git is performed by doing the following:

```bash
if ! hash git 2>/dev/null; then
    echo "************ git not installed, installing ************"
    yum install git -y
fi
```